Compilation breaks on Typescript 4.8.2, Ubuntu 20.04

When running tsc, it breaks with the following errors:
```
root@ec42f57e3eda:/mirrord/vscode-ext# tsc
node_modules/@kubernetes/client-node/dist/informer.d.ts:19:41 - error TS2344: Type 'T' does not satisfy the constraint 'KubernetesObject'.

19     on(verb: string, fn: ObjectCallback<T>): void;
                                           ~

  node_modules/@kubernetes/client-node/dist/informer.d.ts:18:27
    18 export interface Informer<T> {
                                 ~
    This type parameter might need an `extends KubernetesObject` constraint.

node_modules/@kubernetes/client-node/dist/informer.d.ts:20:42 - error TS2344: Type 'T' does not satisfy the constraint 'KubernetesObject'.

20     off(verb: string, fn: ObjectCallback<T>): void;
                                            ~

  node_modules/@kubernetes/client-node/dist/informer.d.ts:18:27
    18 export interface Informer<T> {
                                 ~
    This type parameter might need an `extends KubernetesObject` constraint.

node_modules/@kubernetes/client-node/dist/informer.d.ts:24:106 - error TS2344: Type 'T' does not satisfy the constraint 'KubernetesObject'.

24 export declare function makeInformer<T>(kubeconfig: KubeConfig, path: string, listPromiseFn: ListPromise<T>, labelSelector?: string): Informer<T>;
                                                                                                            ~

  node_modules/@kubernetes/client-node/dist/informer.d.ts:24:38
    24 export declare function makeInformer<T>(kubeconfig: KubeConfig, path: string, listPromiseFn: ListPromise<T>, labelSelector?: string): Informer<T>;
                                            ~
    This type parameter might need an `extends KubernetesObject` constraint.


Found 3 errors in the same file, starting at: node_modules/@kubernetes/client-node/dist/informer.d.ts:19 
```

** Client Version **
0.16.3

** Server Version **
N/A

**To Reproduce**
Steps to reproduce the behavior:
Run tsc on Ubuntu 20.04 with TypeScript 4.8.2

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04
 - NodeJS Version 12.22.9
 
**Additional context**

You can view the error in a wider context in our [CI logs](https://github.com/metalbear-co/mirrord/runs/8057801870?check_suite_focus=true).

Hey, sorry for the delayed response. Running it on OSX this time with Typescript 4.8.3, getting the same errors (in addition to having to add the libraries @yotamselementor mentioned as devDependencies):
```
node_modules/@kubernetes/client-node/dist/informer.d.ts:26:65 - error TS2344: Type 'T' does not satisfy the constraint 'KubernetesObject'.

26     on(verb: ADD | UPDATE | DELETE | CHANGE, cb: ObjectCallback<T>): void;
                                                                   ~

  node_modules/@kubernetes/client-node/dist/informer.d.ts:25:27
    25 export interface Informer<T> {
                                 ~
    This type parameter might need an `extends KubernetesObject` constraint.

node_modules/@kubernetes/client-node/dist/informer.d.ts:28:66 - error TS2344: Type 'T' does not satisfy the constraint 'KubernetesObject'.

28     off(verb: ADD | UPDATE | DELETE | CHANGE, cb: ObjectCallback<T>): void;
                                                                    ~

  node_modules/@kubernetes/client-node/dist/informer.d.ts:25:27
    25 export interface Informer<T> {
                                 ~
    This type parameter might need an `extends KubernetesObject` constraint.

node_modules/@kubernetes/client-node/dist/informer.d.ts:33:106 - error TS2344: Type 'T' does not satisfy the constraint 'KubernetesObject'.

33 export declare function makeInformer<T>(kubeconfig: KubeConfig, path: string, listPromiseFn: ListPromise<T>, labelSelector?: string): Informer<T> & ObjectCache<T>;
                                                                                                            ~

  node_modules/@kubernetes/client-node/dist/informer.d.ts:33:38
    33 export declare function makeInformer<T>(kubeconfig: KubeConfig, path: string, listPromiseFn: ListPromise<T>, labelSelector?: string): Informer<T> & ObjectCache<T>;
                                            ~
    This type parameter might need an `extends KubernetesObject` constraint.


Found 3 errors in the same file, starting at: node_modules/@kubernetes/client-node/dist/informer.d.ts:26
```

Easily worked around with --skipLibCheck though.

--- a/src/informer.ts
+++ b/src/informer.ts
@@ -30,7 +30,7 @@ export type CONNECT = typeof CONNECT;
 export const ERROR = 'error';
 export type ERROR = typeof ERROR;
 
-export interface Informer<T> {
+export interface Informer<T extends KubernetesObject> {
     on(verb: ADD | UPDATE | DELETE | CHANGE, cb: ObjectCallback<T>): void;
     on(verb: ERROR | CONNECT, cb: ErrorCallback): void;
     off(verb: ADD | UPDATE | DELETE | CHANGE, cb: ObjectCallback<T>): void;
@@ -39,7 +39,7 @@ export interface Informer<T> {
     stop(): Promise<void>;
 }
 
-export function makeInformer<T>(
+export function makeInformer<T extends KubernetesObject>(
     kubeconfig: KubeConfig,
     path: string,
     listPromiseFn: ListPromise<T>,

